build_tools/third_party/cuda/CMakeLists.txt

```diff
@@ -5,79 +5,18 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
 function(fetch_cuda_toolkit)
-  # Parameters to the download script.
-  # Look for an appropriate redistrib_*.json here to verify:
-  #   https://developer.download.nvidia.com/compute/cuda/redist/
-  set(_VERSION "11.6.2")
-  set(_PRODUCT "cuda")
-  if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
-    set(_OS "linux")
-  elseif(WIN32)
-    set(_OS "windows")
-  else()
-    message(SEND_ERROR "Unsupported OS environment. Must be Windows or Linux.")
-    return()
+  set(_DOWNLOAD_SCRIPT_PATH "${IREE_SOURCE_DIR}/third_party/nvidia_sdk_download/fetch_cuda_toolkit.py")
+  message(STATUS "Checking for downloaded CUDA toolkit")
+  execute_process(COMMAND ${Python3_EXECUTABLE}
+    "${_DOWNLOAD_SCRIPT_PATH}" "${CMAKE_CURRENT_BINARY_DIR}"
+    RESULT_VARIABLE _EXEC_RESULT
+    OUTPUT_VARIABLE _ACTUAL_DOWNLOAD_PATH
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  if(_EXEC_RESULT AND NOT _EXEC_RESULT EQUAL 0)
+    message(FATAL_ERROR "Error fetching CUDA toolkit")
   endif()
```

Comment:
The first message here looks inaccurate? (It's not _just_ checking, it's actually performing a download)